### DISCUSSION TITLE
Can't verify signature with curve SECP384R1

### DISCUSSION THREAD

**Discussion by matteofumagalli1275** [Infineon/optiga-trust-m#82] [Q&A] [answered]
Hi,

I'm testing out the module with some test function i extracted from mbedtls benchmark code.
If i run the benchmark with the curve SECP256R1 everything works fine, but if i use the curve SECP384R1 the module gets stuck forever in BUSY state during the verify operation.

```c
	mbedtls_ecdsa_context ecdsa;
	const mbedtls_ecp_curve_info *curve_info;
	size_t sig_len;
	int ret;
	unsigned char tmp[200];

	printf("mbedtls_test_sign_and_verify\n");
	
	
	
	for(int i=0; i < 15; i++)
	{
		memset( buf, 0x2A + i, sizeof( buf ) );
		
	      mbedtls_ecdsa_init( &ecdsa );
	      
	      printf("Generate key...\n");
	      if( mbedtls_ecdsa_genkey( &ecdsa, MBEDTLS_ECP_DP_SECP384R1, mbedtls_ctr_drbg_random, NULL ) != 0)
	      {
		      configASSERT(0);
	      }
	      
	      printf("Write signature...\n");
	      if( mbedtls_ecdsa_write_signature( &ecdsa, MBEDTLS_MD_SHA384, buf, 48,
										      tmp, &sig_len, mbedtls_ctr_drbg_random, NULL ) != 0 )
	      {
		      configASSERT(0);
	      }
	      ecp_clear_precomputed( &ecdsa.grp );
	      
	      printf("ECDSA mbedtls_ecdsa_read_signature...\n");
	      
	      ret = mbedtls_ecdsa_read_signature( &ecdsa, buf, 48, tmp, sig_len );
	      
	      printf("ret_verify = %d\n", ret);
	      
	      mbedtls_ecdsa_free( &ecdsa );
	}
```

The mbedtls alternate function were copied from here:
https://github.com/Infineon/optiga-trust-m/blob/develop/examples/mbedtls_port/trustm_ecdsa.c
Except the slot id that was set to 0xE0F1.

Do you have any idea of the cause?

Thanks

**Comment by sgsharath123** [comment, marked as answer]
Hi @matteofumagalli1275 ,

Please check the following hints for probable causes of forever being stuck in a loop:

1. Is OPTIGA initialization sequence (missing in the code snippet above) called before triggering jobs at OPTIGA? I assume it was called for SECP256R1 but not for SECP384R1?
2. If the initialization sequence is called, did it return successfully?
3. It seems like the issue is in synchronization between host and OPTIGA and OPTIGA is either not initialized for communication with host properly or not initialized at all. This could result in OPTIGA being unable to trigger events and callbacks to host.

**Reply by matteofumagalli1275** [reply]
Hi,

The initialization is the same for 256 and 384. It is done out of the snippet code.
I'll check again the initialization and communication porting and see if i find something.

**Reply by matteofumagalli1275** [reply]
Ok i found out.
It was an issue with i2c when sending more then 255 bytes.
